src/library_path.js

```diff
@@ -67,9 +67,7 @@ addToLibrary({
       // EMSCRIPTEN return '/' for '/', not an empty string
       if (path === '/') return '/';
       path = path.replace(/\/+$/g, "");
-      var lastSlash = path.lastIndexOf('/');
-      if (lastSlash === -1) return path;
-      return path.substr(lastSlash+1);
+      return path.split('/').pop();
```

Comment:
This is smaller, but it will allocate garbage, which the old code avoided in at least one case, and even in the other, it only allocates a substring, not an array of substrings. Maybe avoiding allocation is not as critical here as in say the WebGL code, but I'm not sure.

Reply:
I dont't think filesystem operations that involve pathnames are likely to be on critical path are they?  I can see read/write operations maybe caring about this, but I think for string manipulation is probably just prefer size here.

Reply:
@juj WDYT?

Reply:
File system operations in Emscripten are noticeably slow. I would say avoiding the allocation is preferable. Minified it's a difference of 53 bytes.

But if you want something even shorter, how about:
```js
(path) => path && path.match(/([^/]+|\/)\/*$/)[1]
```

Reply:
> Minified it's a difference of 53 bytes.

To be clear, are you saying that because you don't think saving 53 is a worth it?    I think we have make changes for 53 bytes in the past.  Those can add to be real benefits, especially for otherwise-small programs.   However, I guess if I you are using the FS library you program is already probably at least 10s of KiB in the FS library so so big.

Reply:
> But if you want something even shorter, how about:
> 
> ```js
> (path) => path && path.match(/([^/]+|\/)\/*$/)[1]
> ```
> 
> This also fixes the output for `'//'` which should be `'/'` not `''`

I love it!  Thanks

Reply:
I guess the `match` method still generates garbage in the form of two strings but I guess its O(1) and not O(elements in path)

Reply:
Done